test/old-tests/unit/test-kernel-dynamic-tile.cpp

```diff
@@ -0,0 +1,109 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include "RAJA_gtest.hpp"
+
+TEST(KernelDynamicTile, Tile2D) {
+    
+  const int DIM_X = 7;
+  const int DIM_Y = 11;
+  const int TILE_X = 3;
+  const int TILE_Y = 5;
+
+  double* expected = new double[DIM_X * DIM_Y];
+  RAJA::View<double, RAJA::Layout<2> > expectedView(expected, DIM_Y, DIM_X);
+  double* actual = new double[DIM_X * DIM_Y];
+  RAJA::View<double, RAJA::Layout<2> > actualView(actual, DIM_Y, DIM_X);
+
+  using ExecPolicy = RAJA::KernelPolicy< 
+      RAJA::statement::Tile<1, RAJA::tile_dynamic<1>, RAJA::seq_exec,
```

Comment:
This is more of a thought experiment than a comment, but I have a feeling we would do well to have a way to test things like generated ranges and trip counts directly.  This test looks good in that it will test that everything gets toughed and works correctly, but it wont show us whether this actually did what we expect.  I wonder if we could construct a replacement for a Policy level or maybe even a forall that would pass its iterable to its body instead of a single index, so we can directly test the bounds and sizes and so-forth.  Not necessary here, but something I think we might really want to look into for testing policies like this.